test/functional/tests/stress/test_stress_io_small_device.py

```diff
@@ -0,0 +1,107 @@
+#
+# Copyright(c) 2019 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+import pytest
+from datetime import timedelta
+
+from api.cas import casadm, casadm_parser
+from api.cas.cache_config import CacheMode, CacheLineSize, CleaningPolicy
+from storage_devices.disk import DiskType, DiskTypeSet, DiskTypeLowerThan
+from core.test_run import TestRun
+from test_tools.fio.fio import Fio
+from test_tools.fio.fio_param import ReadWrite, IoEngine
+from test_utils.size import Size, Unit
+
+stress_time = timedelta(minutes=30)
+
+
+@pytest.mark.parametrize("cache_mode", [CacheMode.WT, CacheMode.WA, CacheMode.PT])
+@pytest.mark.parametrize("core_number", [1, 4])
+@pytest.mark.parametrize("cache_line_size", CacheLineSize)
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_stress_small_cas_device(cache_mode, cache_line_size, core_number, cleaning_policy=None):
+    """
+        title: Stress test for verifying data on small CAS devices.
+        description: |
+          Validate the ability of the CAS to handle many iops when device is small
+          using different cache modes, cache line sizes and core numbers.
+        pass_criteria:
+          - No system crash.
+          - Md5 sums of core device and cache device are equal.
+    """
+    with TestRun.step(f"Prepare 1 cache and {core_number} core devices."):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([Size(100, Unit.MebiByte)])
+        cache_dev = cache_dev.partitions[0]
+        core_dev = TestRun.disks['core']
+        core_size = []
+        for i in range(core_number):
+            core_size.append(Size(200, Unit.MebiByte))
+        core_dev.create_partitions(core_size)
+
+    with TestRun.step(f"Start cache with {core_number} cores."):
+        cache = casadm.start_cache(cache_dev, cache_mode, cache_line_size, force=True)
+        cores = []
+        for i in range(core_number):
+            cores.append(cache.add_core(core_dev.partitions[i]))
+        caches_count = len(casadm_parser.get_caches())
+        if caches_count != 1:
+            TestRun.fail(f"Expected caches count: 1; Actual caches count: {caches_count}.")
+        cores_count = len(casadm_parser.get_cores(cache.cache_id))
+        if cores_count != core_number:
+            TestRun.fail(f"Expected cores count: {core_number}; Actual cores count: {cores_count}.")
+
+    if cleaning_policy is not None:
+        with TestRun.step("Set cleaning policy."):
+            cache.set_cleaning_policy(cleaning_policy)
+
+    with TestRun.step(f"Stress cache for {int(stress_time.total_seconds() / 60)} minutes."):
+        for core in cores:
```

Comment:
run single fio with 4 jobs (each job should specify different core as target) - consult me if you need help with that